homeassistant/components/smarttub/sensor.py

```diff
@@ -28,3 +45,109 @@ def __init__(self, coordinator, spa):
     def state(self) -> str:
         """Return the current state of the sensor."""
         return self.get_spa_status("state").lower()
+
+
+class SmartTubFlowSwitch(SmartTubEntity):
+    """The state of the flow switch."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "flow_switch")
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self.get_spa_status("flowSwitch").lower()
+
+
+class SmartTubOzone(SmartTubEntity):
+    """The state of the ozone system."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "ozone")
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self.get_spa_status("ozone").lower()
+
+
+class SmartTubBlowoutCycle(SmartTubEntity):
+    """The state of the blowout cycle."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "blowout_cycle")
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self.get_spa_status("blowoutCycle").lower()
+
+
+class SmartTubCleanupCycle(SmartTubEntity):
+    """The state of the cleanup cycle."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "cleanup_cycle")
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self.get_spa_status("cleanupCycle").lower()
+
+
+class SmartTubPrimaryFiltration(SmartTubEntity):
+    """The primary filtration cycle."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "primary_filtration")
+
+    @property
+    def _state(self):
+        return self.get_spa_status("primaryFiltration")
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self._state.get("status").lower()
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        state = self._state
+        return {
+            ATTR_DURATION: state["duration"],
+            ATTR_LAST_UPDATED: state["lastUpdated"],
```

Comment:
Time attributes need to be in isoformat (see `elkm1/sensor.py`)

Reply:
It might already be (please confirm)

Thanks

Reply:
It is (`2021-01-20T11:38:57.014Z`), though I'm returning the raw value from the API. Should we validate it here or is that OK?

Reply:
I could also wrap this in `python-smarttub` and return a Python datetime object, which we would explicitly convert to ISO format. Would that be preferable?

Reply:
Sounds good to me.  Any abstraction that can be moved into the pypi is almost always a win

Reply:
OK, I'll move the two entities with state attributes into a separate PR then.